### ISSUE TITLE
Override CatalogImpl.equals()

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11078]
We must override `CatalogImpl::equals` for performance reasons, to avoid the slow implementation of `AbstractQueryPart::equals`!

A benchmark shows drastic improvement when comparing qualified with partially qualified fields:

```
Benchmark                                                                            Mode  Cnt          Score         Error  Units
BenchmarkRecordAccess.testAccessFirstColumnByExactFieldRef                          thrpt    7  224178590.681 ± 2702249.987  ops/s
BenchmarkRecordAccess.testAccessFirstColumnByIndex                                  thrpt    7  345303597.422 ± 1158450.635  ops/s
BenchmarkRecordAccess.testAccessFirstColumnByName                                   thrpt    7   27200172.142 ±  140554.392  ops/s
BenchmarkRecordAccess.testAccessFirstColumnBySimilarFieldRef                        thrpt    7   15619860.109 ±   73329.193  ops/s
BenchmarkRecordAccess.testAccessFirstColumnBySimilarFieldRefWithTableQualification  thrpt    7     187093.789 ±    1884.166  ops/s


Benchmark                                                                            Mode  Cnt          Score         Error  Units
BenchmarkRecordAccess.testAccessFirstColumnByExactFieldRef                          thrpt    7  182006889.922 ±  421316.931  ops/s
BenchmarkRecordAccess.testAccessFirstColumnByIndex                                  thrpt    7  337722686.377 ± 1146452.959  ops/s
BenchmarkRecordAccess.testAccessFirstColumnByName                                   thrpt    7   26693059.786 ±   72303.230  ops/s
BenchmarkRecordAccess.testAccessFirstColumnBySimilarFieldRef                        thrpt    7   15362907.286 ±   53235.990  ops/s
BenchmarkRecordAccess.testAccessFirstColumnBySimilarFieldRefWithTableQualification  thrpt    7    6637299.749 ±   21159.432  ops/s
```

Benchmark documented here: https://github.com/jOOQ/jOOQ/issues/11058#issuecomment-737312999

**Comment by lukaseder** [comment]
`Object::hashCode` is already overridden efficiently in `AbstractNamed`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.5 (#11079)
